include/irr/asset/IAsset.h

```diff
@@ -152,17 +152,24 @@ class IAsset : virtual public core::IReferenceCounted
 
         virtual core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const = 0;
 
-		friend IAssetManager;
+    protected:
+        void clone_common(IAsset* _clone) const
+        {
+            _clone->m_metadata = m_metadata;
+            _clone->isDummyObjectForCacheAliasing = false;
```

Comment:
copy the `isDummyObjectForCacheAliasing` field or place an `assert(isDummyObjectForCacheAliasing)`